pkg/stanza/docs/operators/json_array_parser.md

```diff
@@ -0,0 +1,145 @@
+## `json_array_parser` operator
+
+The `json_array_parser` operator parses the string-type field selected by `parse_from` with the given header values.
+A JArray string (or a json array string) is a string that represents a JSON array. A JSON array is a type of data structure that is used to store data in a structured way. It consists of an ordered list of values that can be either strings, numbers, objects, or even other arrays.
+#### Examples:
+a simple Jarray string with strictly strings in it:
+```
+"[\"Ford\", \"BMW\", \"Fiat\"]"
+```
+
+Jarray after parsing:
+```json
+["Ford", "BMW", "Fiat"]
+```
+
+a more complex Jarray string with different types in it without nested objects:
+```
+"[\"Hello\", 42, true, null]"
+```
+
+Jarray after parsing:
+```json
+["Hello", 42, true, null]
+```
+
+a more complex Jarray string with different types in it with nested objects:
+```
+"[\"Hello\", 42, {\"name\": \"Alice\", \"age\": 25}, [1, 2, 3], true, null]"
+```
+
+Jarray after parsing:
+```json
+["Hello", 42, {"name": "Alice", "age": 25}, [1, 2, 3], true, null]
+```
+
+Notice that for this example, the current parser will parse every nested object as a string and so the result is actually this - 
+```json
+["Hello", 42, "{\"name\": \"Alice\", \"age\": 25}", "[1, 2, 3]", true, null]
+```
```

Comment:
This has me revisiting [your comment here](https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/30321#issuecomment-1890911017) about potentially creating a separate `headers_matching_parser`.

I still don't think we should change the csv parser (both for stability reasons and because headers are required in csv format). However, I could see value in this parser without any relation to headers.

Then separately, we could have a transformer which, as you suggested, would map an array into an object by providing a key for each index. This could also be useful in other circumstances as well. I would maybe call it `assign_keys` and config would look something like this:

```yaml
# sample config
- type: assign_keys
  keys: [ "foo", "bar" ]
  field: body

# example input
[ 123, "hello" ]

# expected output
{ "foo": 123, "bar": "hello" }
```

What do you think? I know we've iterated on this design quite a bit but I want to get it right and both of these operators could be quite simple.